OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs

```diff
@@ -127,16 +130,36 @@ public class SupportPowerIcon
 
 		public void RefreshIcons()
 		{
-			icons = new Dictionary<Rectangle, SupportPowerIcon>();
-			var powers = spm.Powers.Values.Where(p => !p.Disabled)
-				.OrderBy(p => p.Info.SupportPowerPaletteOrder);
+			var powersHash = spm.PowersHash;
+			var powersModified = false;
+			if (cachedPowers == null || cachedPowersHash != powersHash)
+			{
+				cachedPowers = spm.Powers.Values.OrderBy(p => p.Info.SupportPowerPaletteOrder).ToArray();
+				cachedPowersHash = powersHash;
+				powersModified = true;
+			}
+
+			var hashDisabled = 0;
+			foreach (var p in cachedPowers)
+				if (p.Disabled)
+					hashDisabled ^= p.Key.GetHashCode();
+
+			if (!powersModified && hashDisabled == powersDisabledHash)
+				return;
+
+			powersDisabledHash = hashDisabled;
+
+			icons = new Dictionary<Rectangle, SupportPowerIcon>(cachedPowers.Length);
```

Comment:
Would it make sense to use https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.clear?view=net-5.0 here instead?
`icons.Clear();`